### DISCUSSION TITLE
Loading large file on Uvtools

### DISCUSSION THREAD

**Discussion by sobhonium** [sn4k3/UVtools#721] [Q&A] [answered]
I'm trying to post-process my sliced structure (.png images) with UVtools. Each time I load it on Uvtools, it crashes. I can see on the title bar that the memory usage exceeds 6.5GB and it crashes afterwards! For smaller files it works fine but for a larger file like this try it won't. I think UVtools loads the entire structure for its computations (resin traps/islands/...).  Is there anyways to help UVtools do its job without crashing?

The file I'm loading is a .pwmb file (Anycube Mono X file format) which includes 3200 layers (images) for 15cm-height structure.

That's my Memory space info: 
![Screenshot from 2023-06-07 10-23-36](https://github.com/sn4k3/UVtools/assets/129010715/049d0fa4-39df-40d5-a351-743db195e4ea)

**Comment by sn4k3** [comment]
It crashes while on load the file or after with computations?
Can you provide the file?

**Reply by sobhonium** [reply]
Loading is long but is OK. Computations is here to blame.
Sure here it is one of those files I'm tesing:
[file1.zip](https://github.com/sn4k3/UVtools/files/11670896/file1.zip)

**Reply by sn4k3** [reply, marked as answer]
I was able to compute that file issues with a peak of near 10gb ram in only 39s
```
#	Started	Time(s)	Description
5	00:17:36	38.99	Computing Issues
```


What is your CPU?
The more CPU core count you have more RAM you need to have to keep the pace.

If you hitting RAM limit you need to reduce workers on UVtools settings or use other layer compression

**Comment by sobhonium** [comment]
I have a 4-processor CPU:
Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz
cache size	: 6144 KB
cpu cores	: 4

Anyways, Changing "Settings+Thread from -1/4 to 2/4 and + Compression=High " worked. Thanks.

**Reply by sn4k3** [reply]
Linux may have poor memory management when it comes to hit limits and using the swap in this case (Also your swap is too small). Using less workers will always reduce the memory usage as each worker is decoding/encoding an image and processing data. In these days even 32GB seens small (In my case) where system idle already takes 9gb
